lib/video_merger.ex

```diff
@@ -52,73 +54,30 @@ defmodule Membrane.VideoMerger do
     {:ok, Map.put_new(state, id, [])}
   end
 
-  @impl true
-  def handle_process(pad, buffer, ctx, state), do: handle_process_list(pad, [buffer], ctx, state)
-
   @impl true
   def handle_process_list({_pad, :input, id}, buffers, _context, state) do
     if not Enum.all?(buffers, &Map.has_key?(&1.metadata, :pts)) do
       raise("Cannot merge stream without pts")
     end
 
     state
-    |> Map.update!(id, &(&1 ++ buffers))
+    |> BufferQueue.enqueue_list(id, buffers)
     |> get_actions(redemand: :output)
   end
 
-  defp get_empty_inputs(state) do
-    state
-    |> Enum.filter(fn {_id, buffers} -> buffers == [] end)
-    |> Enum.map(fn {id, _val} -> id end)
-  end
-
   defp get_actions(state, fallback) do
-    case get_buffers(state) do
-      {[], new_state} when new_state == %{} ->
-        {{:ok, [end_of_stream: :output] ++ fallback}, %{}}
+    case BufferQueue.dequeue_buffers(state) do
+      {:ok, :end_of_stream} ->
+        {{:ok, fallback ++ [end_of_stream: :output]}, %{}}
 
-      {buffers, new_state} when new_state == %{} ->
-        {{:ok, [end_of_stream: :output, buffers: buffers] ++ fallback}, %{}}
+      {{:ok, buffers: buffers}, :end_of_stream} ->
+        {{:ok, [buffers: buffers] ++ fallback ++ [end_of_stream: :output]}, %{}}
```

Comment:
This action is invalid: it's `:buffer` not `:buffers` and it is missing pad name before buffers, this is why the test I mentioned earlier ([0-1] and [1-2] on cutters) fails. I guess test cases should include that
```suggestion
        {{:ok, [buffer: {:output, buffers}] ++ fallback ++ [end_of_stream: :output]}, %{}}
```